tagstore/backends/tagstore_django/admin.py

```diff
@@ -0,0 +1,80 @@
+""" Admin for tags. """
+
+from django import forms
+from django.contrib import admin
+
+from tagstore.backends.django import DjangoTagstore
+# from tagstore.models import EntityId
+
+from .models import Taxonomy, Tag, Entity, MAX_CHAR_FIELD_LENGTH
+
+
+class CustomTagAdminForm(forms.ModelForm):
+    """ Sets an extra field `parent` which does not exist in Tag. """
+    parent = forms.CharField(max_length=MAX_CHAR_FIELD_LENGTH, required=False)
+
+    def __init__(self, *args, **kwargs):
+        """ Adds the parent name, if it exists, to the edit tag form. """
+        parent_tag = ''
+        if 'instance' in kwargs and kwargs['instance'] is not None:
+            if kwargs['instance'].parent_tag_tuple:
+                parent_tag = kwargs['instance'].parent_tag_tuple.name
+        super(CustomTagAdminForm, self).__init__(*args, **kwargs)
+        self.fields['parent'].widget.attrs.update({'value': parent_tag})
+
+    # TODO: create a dry run flag for add_tag_to_taxonomy which
+    # can be used to validate before attempting to save
+
+
+class TagAdmin(admin.ModelAdmin):
+    """ Controls display and saving of Tag model objects. """
+    list_filter = ('taxonomy',)
+    list_display = ('name', 'path', 'taxonomy')
```

Comment:
I can't currently access the `list` views, so these don't seem to be having any effect?

Reply:
I wasn't sure the project's stance on avoiding dead code vs leaving useful scaffolding for features we'll likely want later, so thanks for clarifying. I have taken that out.